tests/basic.js

```diff
@@ -1,44 +1,117 @@
 import StyleObserver from "../index.js";
-import gentleRegisterProperty from "../src/util/gentle-register-property.js";
-import tests from "./tests.js";
+import gentleRegisterProperty from "./util/gentle-register-property.js";
+import types from "./util/types.js";
+
+let dummy;
 
 export default {
 	name: "Basic",
 
-	run ({property, meta, initial, value}) {
-		let dummy = document.createElement("div");
+	map (value) {
+		if (this.data.property) {
+			dummy.style.setProperty(this.data.property, value);
+			let computed = getComputedStyle(dummy).getPropertyValue(this.data.property);
+			console.log(computed, value);
+			return computed;
+		}
+
+		return value;
+	},
+
+	beforeAll () {
+		dummy = document.createElement("div");
 		document.body.append(dummy);
+	},
 
-		let observer;
-		return new Promise((resolve, reject) => {
-			observer = new StyleObserver(records => {
-				resolve(records);
-			});
-
-			if (meta) {
-				gentleRegisterProperty(property, meta);
-			}
-
-			if (initial) {
-				dummy.style.setProperty(property, initial);
-			}
-
-			observer.observe(dummy, property);
-
-			dummy.style.setProperty(property, value);
-
-			// Timeout after 500ms
-			setTimeout(() => reject(), 500);
-		})
-		.catch(_ => {
-			return [{ value: "Timed out" }];
-		})
-		.then(records => records[0].value)
-		.finally(() => {
-			observer.unobserve(dummy);
-			dummy.remove();
+	beforeEach () {
+		let element = document.createElement("div");
+		document.body.append(element);
+
+		let result = {};
+		result.promise = new Promise((resolve, reject) => {
+			result.resolve = resolve;
+			result.reject = reject;
+		});
+
+		let observer = new StyleObserver(records => {
+			let value = records[0].value;
+			result.resolve(value);
 		});
+
+		Object.assign(this.data, { element, observer, result });
 	},
 
-	tests,
+	run (property, value, initialValue) {
+		let { observer, element, result } = this.data;
+
+		element.style.setProperty(property, initialValue);
+		observer.observe(element, property);
+		element.style.setProperty(property, value);
+
+		setTimeout(() => result.reject("Timeout"), 500);
+
+		return result.promise;
+	},
+
+	afterEach () {
+		this.data.observer.unobserve(this.data.element);
+		this.data.element.remove();
+	},
+
+	afterAll () {
+		dummy.remove();
+	},
+
+	tests: [
+		{
+			name: "Built-in properties",
+			tests: Object.entries(types).map(([id, meta]) => {
```

Comment:
We should probably filter out tests that don't have `meta.property`. Otherwise, those tests will always fail (for now, we have two: with syntax `*` and `<time>`).

Reply:
Yup, sounds good.